### ISSUE TITLE
Bug: overwriteRequestSecurity removes API key when no key is passed

### ISSUE THREAD

**Issue by savage-alex** [apideck-libraries/portman#468]
When using overwriteRequestSecurity within a variation test,
Given that an overwrite is applied which changes the apiKey value BUT does not include "key" (the name of the API key)
![image](https://user-images.githubusercontent.com/25430683/223744134-971d025b-0e65-4e43-b789-0871f0cb0674.png)

Then the created postman collection has the api key "Key" is missing
![image](https://user-images.githubusercontent.com/25430683/223744177-ce54769c-5f8f-4991-a43f-dcb10e57ea16.png)

Example config:
```
{
    "version": 1.0,
    "tests": {
        "contractTests": [
            {
                "openApiOperation": "*::/*",
                "statusCode": {
                    "enabled": true
                }
            },
            {
                "openApiOperation": "*::/*",
                "responseTime": {
                    "enabled": true,
                    "maxMs": 2000
                }
            },
            {
                "openApiOperation": "*::/*",
                "contentType": {
                    "enabled": true
                }
            },
            {
                "openApiOperation": "*::/*",
                "jsonBody": {
                    "enabled": true
                }
            },
            {
                "openApiOperation": "*::/*",
                "schemaValidation": {
                    "enabled": true,
                    "additionalProperties": false
                }
            },
            {
                "openApiOperation": "*::/*",
                "headersPresent": {
                    "enabled": true
                }
            }
        ],
        "contentTests": [],
        "extendTests": [],
        "variationTests": [
            {
                "openApiOperation": "*::/*",
                "openApiResponse": "401",
                "variations": [
                    {
                        "name": "noAPIKey",
                        "overwrites": [
                            {
                                "overwriteRequestSecurity": {
                                    "apiKey": {
                                        "value": ""
                                    }
                                }
                            }
                        ],
                        "tests": {
                            "contractTests": [
                                {
                                    "statusCode": {
                                        "enabled": true
                                    },
                                    "schemaValidation": {
                                        "enabled": true,
                                        "additionalProperties": false
                                    },
                                    "contentType": {
                                        "enabled": true
                                    }
                                }
                            ]
                        }
                    }
                ]
            }
        ]
    }
}
```

**Comment by savage-alex** [comment]
the docs suggest that Key is optional but unless it is included like in this config the Key is missing from postman:

Working config:
```
{
    "version": 1.0,
    "tests": {
        "contractTests": [
            {
                "openApiOperation": "*::/*",
                "statusCode": {
                    "enabled": true
                }
            },
            {
                "openApiOperation": "*::/*",
                "responseTime": {
                    "enabled": true,
                    "maxMs": 2000
                }
            },
            {
                "openApiOperation": "*::/*",
                "contentType": {
                    "enabled": true
                }
            },
            {
                "openApiOperation": "*::/*",
                "jsonBody": {
                    "enabled": true
                }
            },
            {
                "openApiOperation": "*::/*",
                "schemaValidation": {
                    "enabled": true,
                    "additionalProperties": false
                }
            },
            {
                "openApiOperation": "*::/*",
                "headersPresent": {
                    "enabled": true
                }
            }
        ],
        "contentTests": [],
        "extendTests": [],
        "variationTests": [
            {
                "openApiOperation": "*::/*",
                "openApiResponse": "401",
                "variations": [
                    {
                        "name": "noAPIKeyValue",
                        "overwrites": [
                            {
                                "overwriteRequestSecurity": {
                                    "apiKey": {
                                        "key" : "x-api-key",
                                        "value": ""
                                    }
                                }
                            }
                        ],
                        "tests": {
                            "contractTests": [
                                {
                                    "statusCode": {
                                        "enabled": true
                                    },
                                    "schemaValidation": {
                                        "enabled": true,
                                        "additionalProperties": false
                                    },
                                    "contentType": {
                                        "enabled": true
                                    }
                                }
                            ]
                        }
                    },
                    {
                        "name": "invalidAPIKeyValue",
                        "overwrites": [
                            {
                                "overwriteRequestSecurity": {
                                    "apiKey": {
                                        "key" : "x-api-key",
                                        "value": "foobar"
                                    }
                                }
                            }
                        ],
                        "tests": {
                            "contractTests": [
                                {
                                    "statusCode": {
                                        "enabled": true
                                    },
                                    "schemaValidation": {
                                        "enabled": true,
                                        "additionalProperties": false
                                    },
                                    "contentType": {
                                        "enabled": true
                                    }
                                }
                            ]
                        }
                    }

                ]
            }
        ]
    }
}
```

**Comment by thim81** [comment]
hi @savage-alex 

We found the cause of the bug and created a [PR](#473), that after review will be included in the next release.
Thanks again for taking the time to provide us with all the details of the bug.

**Comment by savage-alex** [comment]
Thank you @thim81

**Comment by thim81** [comment]
Hi @savage-alex 

We just released Portman v1.22.0, which tackles the reported behavior of `overwriteRequestSecurity`.
It should solve your issue.
